mythril/laser/ethereum/state/annotation.py

```diff
@@ -34,6 +36,23 @@ def persist_over_calls(self) -> bool:
         """
         return False
 
+    def check_merge_annotation(self, annotation) -> bool:
```

Comment:
Shouldn't this be part of the MergeableStateAnnotation class?

Reply:
I added it here because StateAnnotation cannot be merged.

Reply:
This adds an implicit dependency from `StateAnnotation` to `MergeableStateAnnotation`. 

It might be cleaner if callers of `check_merge_annotation()` would do a check like:`annotation instanceof MergeableStateAnnotation and annotation.check_merge_annotation(other)`